crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -453,41 +466,78 @@ impl Format<PyFormatContext<'_>> for FormatNormalizedComment<'_> {
 fn normalize_comment<'a>(
     comment: &'a SourceComment,
     source: SourceCode<'a>,
-) -> FormatResult<Cow<'a, str>> {
+) -> FormatResult<NormalizedComment> {
```

Comment:
I think this code became a little bit less clear due to the use of `Cursor`. It would be much clearer if we didn't have the `&nbsp` special-casing.